### PR TITLE
BIP68: Simplify language and update for current implementation

### PR DESCRIPTION
**Pull request by btcdrak**
Update according to bitcoin/bitcoin#6312 implementation

### CODE REVIEW

**Comment by morcos** [comment]
I took a quick skim, the example code in Compatibility is still referring to the old specification.

My biggest outstanding concern is still with this flag at (1<<22).  I'd like to make sure a few other people take a look at it and agree it makes sense.  I think the one justification @maaku had for this is it would allow alternative coins/chains with faster block times to have finer granularity than 512 seconds and work out of the box.   I'm not sure how much weight should be given to that, but on the other hand, it may be bike shedding at this point as to where exactly we put the flag and that may be a perfectly reasonable choice.   So concept ACK from me.

**Comment by btcdrak** [comment]
@morcos Thanks for taking a look. The compatibility section has been updated now. I wasnt sure if I should mention the granularity issue in relation to shorter block intervals and sidechains because it is not directly related to Bitcoin but dont have a problem adding something it is deemed important.

Overall I have tried to explain rationale as much as possible and make the BIP clearer.

The main rationale about (1<<22) I believe was taking into consideration 3 byte pushdata when using an opcode like OP_CSV. There is a paragraph to explain this. This comes from the discussion here http://bitcoinstats.com/irc/bitcoin-dev/logs/2015/10/15#l1444928045.0

**Comment by jmcorgan** [comment]
utACK

File: bip-0068.mediawiki [hunk 1]
```diff
@@ -9,19 +12,40 @@
...
+Bitcoin transactions have a sequence number field for each input. The original idea appears to have been that a transaction in the mempool would be replaced by using the same input with a higher sequence value. Although this was not properly implemented, it assumes miners would prefer higher sequence numbers even if the lower ones were more profitable to mine. However, a miner acting on profit motives alone would break that assumption completely. The change described by this BIP repurposes the sequence number for new use cases without breaking existing functionality. It also leaves room for future expansion and other use cases.
+
+The transaction nLockTime is used to prevent the mining of a transaction until a certain date. (in block height or time)
```
Comment (instagibbs) [inline bip-0068.mediawiki]: drop the parenthesis, just add "in block height or time" to the sentence

**Comment by btcdrak** [comment]
Added an encoding diagram for clarity.

**Comment by CodeShark** [comment]
@btcdrak Beautiful! So much more clear than before. ACK

**Comment by ajtowns** [comment]
The (1<<22) rationale is still a bit confusing to me... Maybe something like:

"The flag bit (1<<22) was chosen as the highest bit in a positive 3-byte signed integer. This allows its value to be constructed in bitcoin scripts for use with OP_CHECKSEQUENCEVERIFY (BIP 112), either by setting it directly in a straighforward manner with a 3-byte PUSHDATA, or calculating it using signed arithmetic operators."

would be better?

Otherwise, utACK fwiw.

**Comment by btcdrak** [comment]
@ajtowns I simplified the sentence. Thanks.

**Comment by NicolasDorier** [comment]
Sounds good to me, ACK

**Comment by instagibbs** [comment]
Carefully went through the latest draft of the BIP, and checked the related sections in the Core pull, especially LockTime(). Description and code lines up.

**Comment by rustyrussell** [comment]
Ack.  This is a nice, clear change, and meets my understanding of consensus last it was discussed.

**Comment by luke-jr** [comment]
This needs an ACK from @maaku

**Comment by btcdrak** [comment]
@luke-jr As you are aware @maaku stepped down from championing the proposals and handed them over to me. Its unfair to expect him to ACK this, although it would be nice if he did.

bitcoin/bitcoin#6312 is the implementation. The current master branch of BIP68 now **does not match** the implementation which the BIP points to. This is causing serious confusion. A total of three people in the last week have come asking me questions about bitcoin/bitcoin#6312 because they read the BIP and it doesn't seem to match the implementation at bitcoin/bitcoin#6312.

I might add that the implementation was updated by @maaku before he left and handed this over, however I forgot to update the text.

These modifications have been checked by @ajtowns @rustyrussell @instagibbs and @CodeShark, surely those are enough ACKs :-P

**Comment by luke-jr** [comment]
@btcdrak This PR includes the change of champion, so the current champion needs to ACK it. (I also have only heard of the handover from third-parties, so I don't even have personal first-hand knowledge of this.)

**Comment by btcdrak** [comment]
@luke-jr It is unfortunate that the conversations were in #bitcoin-core-dev a few days before we got channel logging setup iirc.

**Comment by maaku** [comment]
I unconditionally ACK the new list of champions.

Separately, having read the new text I suggest that acronyms be expanded. There's a couple of points where acronyms like RTL and MTP are used, even without expansion on first use. Standards should be clear and unambiguous, even if that means a little verbosity.

**Pull request merged**